embulk-jruby-strptime/src/main/java/org/embulk/spi/time/StrptimeParser.java

```diff
@@ -0,0 +1,883 @@
+package org.embulk.spi.time;
+
+import java.io.IOException;
+import java.io.Reader;
+import java.io.StringReader;
+import java.util.EnumSet;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Arrays;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.embulk.spi.time.lexer.StrptimeLexer;
+
+/**
+ * This is Java implementation of ext/date/date_strptime.c in Ruby v2.3.1.
+ * @see https://github.com/ruby/ruby/blob/394fa89c67722d35bdda89f10c7de5c304a5efb1/ext/date/date_strptime.c
+ *
+ * TODO
+ * This class is tentatively required for {@code TimestampParser} class.
+ * After embulk-jruby-strptime will be merged into JRuby, embulk-jruby-strptime
+ * will be removed from Embulk.
```

Comment:
Ok, let's keep in Embulk. `StrptimeParser` class that we contributed to JRuby is `public` modifier but, that's kinds of internal class from view point of almost all of JRuby users. Therefore, the class possibly might be refactored dramatically (name and package as well).